roles/bootstrap/tasks/main.yml

```diff
@@ -104,11 +105,24 @@
     name: "{{ bootstrap_default_common_packages }}"
     state: present
 
+- name: Check if reboot required
+  ansible.builtin.stat:
+    path: /var/run/reboot-required
+  register: reboot_required_file
+
 # Disable swap
 - name: Disable SWAP in fstab
   ansible.builtin.replace:
     path: /etc/fstab
     regexp: '^([^#].*?\sswap\s+sw\s+.*)$'
     replace: '# \1'
   when: bootstrap_swap_disable
-  notify: Disable SwAP
+  register: bootstrap_swap_disabled
+  notify: Disable SWAP
+
+- name: Force all notified handlers to run at this point, not waiting for normal sync points (required to be triggered before reboot)
+  ansible.builtin.meta: flush_handlers
```

Comment:
Instead of always flashing all handlers, maybe it would make sense to do this only when a reboot is required. So something like:


```yaml
- name: Check if reboot required
  ansible.builtin.stat:
    path: /var/run/reboot-required
  register: bootstrap_reboot_required_file
  
- name: Set reboot required
  ansible.builtin.set_fact:
    bootstrap_reboot_required: bootstrap_reboot_required_file.stats.exists
 
- name: Prompt for reboot
  ansible.builtin.pause:
    prompt: "A reboot is required. Do you want to proceed? (y/n)"
  when: bootstrap_reboot_required
  register: bootrap_reboot_prompt

- name: Handle reboot
  when: bootstrap_reboot_required and bootrap_reboot_prompt == "y"
  block:
  - name: Flush handlers
    ansible.builtin.meta: flush_handlers
  - name: Reboot
    ansible.builtin.reboot:
```

It also uses the `pause` module (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pause_module.html ) so that we can only prompt the user whenever there is really a reboot required instead of doing it always over the `vars_prompt`.